.github/workflows/codecov.yml

```diff
@@ -14,36 +14,19 @@ on:
 jobs:
   run:
     runs-on: self-hosted
+    container: 
+      image: towhee/towhee-ut:latest
     steps:
-    - uses: actions/checkout@v3.0.0
-    - name: Setup Python
-      uses: actions/setup-python@v3.0.0
-      with:
-        python-version: 3.8
     - name: Generate coverage report
       run: |
-        source /test/miniconda3/etc/profile.d/conda.sh
-        conda env remove -n towhee-ut
-        conda create -n towhee-ut python=3.8
-        conda activate towhee-ut
-        conda install pytorch torchvision torchaudio cudatoolkit=11.3 -c pytorch
-        conda run -n towhee-ut pip install coverage pytest pytest-xdist
-        conda run -n towhee-ut pip install -r test_requirements.txt
-        conda run -n towhee-ut coverage erase
-        conda run -n towhee-ut coverage run -m pytest
-    - name: Generate xml report
-      run: |
+        rm -rf towhee
+        git clone https://github.com/towhee-io/towhee.git
```

Comment:
I think it is better to use `actions/checkout` instead of using the git command line directly. The action should be able to checkout the right branch for testing.

Reply:
Also, by clone repo every time, we are running ut on main branch, so the PR is not tested, we need to change the way we load tests dir.